packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,601 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import {
+  memo,
+  useMemo,
+  useState,
+  MouseEvent,
+  ReactElement,
+  ReactNode,
+  useCallback,
+} from "react";
+
+import { Box } from "../Box";
+import { Button } from "../Button";
+import type { HtmlProps } from "../HtmlProps";
+import { CollapseLeftIcon, ExpandLeftIcon } from "../icons.generated";
+import { Link } from "../Link";
+import { NavAccordion } from "./NavAccordion";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../OdysseyDesignTokensContext";
+import { Status, statusSeverityValues } from "../Status";
+import { Heading6 } from "../Typography";
+
+export type SideNavItem = {
+  id: string;
+  label: string;
+  /**
+   * The icon element to display at the end of the Nav Item
+   */
+  endIcon?: ReactElement;
+  /**
+   * Whether the item is expanded by default
+   */
+  isDefaultExpanded?: boolean;
+  /**
+   * Whether the item is expanded
+   */
+  isExpanded?: boolean;
+  /**
+   * Whether the item is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether the item is active/selected
+   */
+  isSelected?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(event: MouseEvent<HTMLAnchorElement>): void;
+  /**
+   * The status element to display after the label
+   */
+  severity?: (typeof statusSeverityValues)[number];
+  /**
+   * The icon element to display at the start of the Nav Item
+   */
+  startIcon?: ReactElement;
+  /**
+   * The label to display inside the status
+   */
+  statusLabel?: string;
+  target?: string;
+} & (
+  | {
+      /**
+       * Determines if the side nav item is a section header
+       */
+      isSectionHeader: true;
+      href?: never;
+      children?: never;
+    }
+  | {
+      isSectionHeader?: false;
+      href: string;
+      /**
+       * An array of side nav items
+       */
+      children?: SideNavItem[];
+    }
+);
```

Comment:
How do the docs show up for this? Bryan is doing a lot of with prop types and documentation; he might know more about the specifics of how docs look in this pattern.

I wanna make sure we're documenting each prop, and it shows up the way you'd expect.

Reply:
![image](https://github.com/okta/odyssey/assets/40774045/8b871d44-963e-400a-90b8-8a5e4564ce33)
![image](https://github.com/okta/odyssey/assets/40774045/65e42f62-dc60-4e41-8748-e5386a5da7b5)